src/Native/Unix/System.Native/pal_io.h

```diff
@@ -267,12 +267,22 @@ enum SysConfName
  */
 enum PollEvents
 {
+#if defined(_AIX)
+/* of course AIX is different */
+    PAL_POLLIN = 0x0001,   /* non-urgent readable data available */
+    PAL_POLLPRI = 0x0004,  /* urgent readable data available */
+    PAL_POLLOUT = 0x0002,  /* data can be written without blocked */
+    PAL_POLLERR = 0x4000,  /* an error occurred */
+    PAL_POLLHUP = 0x2000,  /* the file descriptor hung up */
+    PAL_POLLNVAL = 0x8000, /* the requested events were invalid */
+#else
```

Comment:
This it not correct.  These values are the PAL values and must be identical to the ones used in the managed definition:
https://github.com/dotnet/corefx/blob/d9d06b0109c600cd7a13a8fa5783d8e6b0c745a4/src/Common/src/Interop/Unix/System.Native/Interop.Poll.cs#L14-L23

You will need to create the appropriate mapping when these are used.

Reply:
Would the mapping need to be implemented on the C# side (figure out a way to ifdef on it) or translate the system values to the PAL ones et vice versa?

Reply:
No, on the native side.  The idea is that the managed code always uses a fixed set of values (the platform abstraction layer "PAL" values) when communicating with the native shim.  The native shim is then compiled on each platform where there are differences, and it's responsible for ensuring that the managed code is always able to use those PAL values, so if the PAL values aren't the same such that it can't just pass them through, the native code needs to do the translation.